app/controllers/paginable/research_outputs_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Paginable
+  # Controller for paginating/sorting/searching the research_outputs table
+  class ResearchOutputsController < ApplicationController
+    include Paginable
+
+    after_action :verify_authorized
+
+    # GET /paginable/plans/:plan_id/research_outputs
+    def index
+      @plan = Plan.find_by(id: params[:plan_id])
+      # Same @research_outputs assignment as app/controllers/research_outputs_controller.rb
+      @research_outputs = ResearchOutput.includes(:repositories).where(plan_id: @plan.id)
+      # Same authorize handling as app/controllers/research_outputs_controller.rb
+      authorize @research_outputs.first || ResearchOutput.new(plan_id: @plan.id)
```

Comment:
In the case of `app/controllers/research_outputs_controller.rb`:
If I only use `authorize @research_outputs.first`, I encounter the following if the plan has no research_outputs:
```
Pundit::NotDefinedError - unable to find policy `NilClassPolicy` for `nil`:
  app/controllers/research_outputs_controller.rb:17:in `index'
```
If I use `authorize @research_outputs.first if @research_outputs.first.present?`, I encounter the following if the plan has no research_outputs:
```
Pundit::AuthorizationNotPerformedError at /plans/14224/research_outputs
ResearchOutputsController
```

However, it may be the case that there is always at least one research output when `app/controllers/paginable/research_outputs_controller.rb` is called. But it might still be safest to keep `authorize @research_outputs.first || ResearchOutput.new(plan_id: @plan.id)`? (I could at least add a comment explaining this a bit).

Reply:
A comment would be good. Also add the method call parenthesis to clarify the expected order.

Reply:
> However, it may be the case that there is always at least one research output when `app/controllers/paginable/research_outputs_controller.rb` is called. But it might still be safest to keep `authorize @research_outputs.first || ResearchOutput.new(plan_id: @plan.id)`? (I could at least add a comment explaining this a bit).

I guess `|| ResearchOutput.new(plan_id: @plan.id)` prevents `Pundit::NotDefinedError` when a direct GET /paginable/plans/:id/research_outputs request is made on a plan with 0 research_outputs.

Reply:
> A comment would be good. Also add the method call parenthesis to clarify the expected order.

Cool, I made one more commit addressing this.